kura/org.eclipse.kura.core.configuration/src/main/java/org/eclipse/kura/core/configuration/ConfigurationServiceImpl.java

```diff
@@ -577,7 +577,12 @@ public synchronized void rollback(long id) throws KuraException {
 
     @Override
     public List<ComponentConfiguration> getSnapshot(long sid) throws KuraException {
+        List<ComponentConfiguration> returnConfigs = new ArrayList<ComponentConfiguration>();
+
         XmlComponentConfigurations xmlConfigs = loadEncryptedSnapshotFileContent(sid);
+        if (xmlConfigs == null) {
```

Comment:
It's not an uncommon practice in this service and reduces indentation level of the main part of the method, but that's just fine, too. Will update.